optimus/io/abstract_driver.py

```diff
@@ -0,0 +1,56 @@
+from abc import abstractmethod, ABC
+from enum import Enum
+
+
+class AbstractDriver(ABC):
+    """Base class for all database implementations"""
+
+    @abstractmethod
+    def properties(self) -> Enum:
+        """
+        Returns a enum containing the following database properties:
+
+            * name
+            * port
+            * java_class
+            * table_name
+
+        :return: en enum of type ``DriverProperties``
+        """
+        pass
+
+    @abstractmethod
+    def url(self, *args, **kwargs) -> str:
+        """
+        Returns a string connection to the underlying database implementation.
+        :param conn_properties: connection properties
+        :return: the connection string
+        """
+        pass
+
+    @abstractmethod
+    def table_names_query(self, *args, **kwargs) -> str:
+        """
+        Returns the query for all tables in the database.
+        :param query_parameters: query parameters
+        :return: a query for all the tables in a given database
+        """
+        pass
+
+    @abstractmethod
+    def table_name_query(self, *args, **kwargs) -> str:
+        """
+        Returns the query for all tables in the database.
```

Comment:
The description is the same that ```def table_names_query```. Is this correct?

Reply:
Good catch, fixed